SwissKnife/avro/AvroTransformer.py

```diff
@@ -0,0 +1,271 @@
+        transform_name can be directly the name of a function or it can be
+        a expresion, so the result will be a closure. For example,
+        "copyFrom(date)" refers to function "copyFrom" whith the parameter "date".
+        The returned function has two parameters:
+            - value: The current value of the field.
+            - record: The entire record.
+
+        Available functions:
+            - int2boolean: Transform a int to a bolean. True if x > 0, otherwise False.
+            - copyFrom(y): Return the value of field Y. So, the current value of the
+                           target field is ignored.
+
+        :param transform_name: The transform name.
+        :type transform_name: str
+        :raises RuntimeError: Raises Exception if transform_name is invalid.
+        :raises RuntimeError: copyFrom function raises Exception when it is executed
+                              if the provided field is invalid.
+        :return: A transform function.
+        :rtype: Callable[[object, dict], object]
+        """
+        selected_function = None
+        if transform_name == "int2boolean":
+            def int2boolean(value: int, record: dict) -> bool:
+                if value is None:
+                    return None
+                else:
+                    return int(value) > 0
+            selected_function = int2boolean
+        elif re.match("copyFrom\(\w+\)", transform_name):
+            copied_field = re.match("copyFrom\((\w+)\)", transform_name)[1]
+
+            def copyFrom(value:int, record:dict) -> bool:
+                if copied_field in record:
+                    return record[copied_field]
+                else:
+                    raise RuntimeError("Invalid field in copyFrom")
+
+            selected_function = copyFrom
+        # If there is not an available function, raise Exception
+        if selected_function is None:
+            raise RuntimeError("Invalid name for a transform function")
+        return selected_function
+
+    def get_record_with_casted_values(self, record: Record) -> Record:
+        new_record = {}
+        for key, value in record.items():
+            try:
+                casted_value = self._get_casted_value(key, value, self.cast_dict)
+            except Exception as ex:
+                raise ValueError(f"In the record {record} the next error {ex} in the field {key} with value {value}")
+            new_record[key] = casted_value
+        return new_record
+
+    @staticmethod
+    def _get_casted_value(key: str, value: object, cast_dict: dict) -> object:
+        types_list = cast_dict[key]
+        for type_to_cast in types_list:
+            if type_to_cast == "null" and value is None:
+                return None
+            elif type_to_cast in ["int", "long"]:
+                # if empty string
+                if type(value) is str and len(value) == 0:
+                    return None
+                else:
+                    return int(value)
```

Comment:
Could we have problems when casting really big `long` values? I guess it's not a problem we are facing now nor should correct right now, I was just wondering...

Reply:
According to [python documentation](https://docs.python.org/3.6/library/stdtypes.html#typesnumeric): 

> There are three distinct numeric types: integers, floating point numbers, and complex numbers. [...] **Integers have unlimited precision.** [...]

So in this case, I think is not an issue.

Reply:
cool! thanks for the link :+1: